lib/sync/fetcher_test.go

```diff
@@ -0,0 +1,72 @@
+package sync
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"boscoin.io/sebak/lib/block"
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/network"
+	"boscoin.io/sebak/lib/node"
+	"boscoin.io/sebak/lib/node/runner"
+	"boscoin.io/sebak/lib/storage"
+	"github.com/stellar/go/keypair"
+	"github.com/stretchr/testify/require"
+)
+
+func renderNodeItem(w http.ResponseWriter, itemType runner.NodeItemDataType, o interface{}) {
+	s, err := json.Marshal(o)
+	if err != nil {
+		itemType = runner.NodeItemError
+		s = []byte(err.Error())
+	}
+
+	writeNodeItem(w, itemType, s)
+}
+
+func writeNodeItem(w http.ResponseWriter, itemType runner.NodeItemDataType, s []byte) {
+	w.Write(append([]byte(itemType+" "), append(s, '\n')...))
+}
+
+func TestBlockFetcher(t *testing.T) {
+	st := storage.NewTestStorage()
+	defer st.Close()
+
+	kp, _ := keypair.Random()
+	_, nw, localNode := network.CreateMemoryNetwork(nil)
+	cm := &mockConnectionManager{
+		allConnected: []string{kp.Address()},
+		getNodeFunc: func(addr string) node.Node {
+			ep, err := common.NewEndpointFromString("https://node1?NodeName=n1")
+			require.Nil(t, err)
+			v, err := node.NewValidator(kp.Address(), ep, "n1")
+			require.Nil(t, err)
+			return v
+		},
+	}
+
+	bk := block.TestMakeNewBlock([]string{})
+	bk.Height = uint64(1)
```

Comment:
`block.InitTestBlockchain`

Reply:
The test setup is wrong without InitTestBlockchain?

Reply:
It's incomplete.
It will work for this case (obviously) but as soon as someone decides to change the rules of one function, it will break. `InitTestBlockchain` was introduced to provide a single function that "does the right thing".
This way, if we decide to, for example, add an extra operation to the genesis block, we only have one function to change, instead of going through all the failing test cases.

Reply:
Ok l got it